scripts/create-agent.sh

```
@@ -0,0 +1,640 @@
+        if [[ "$LOCK_ACQUIRED" == "flock" ]]; then
+            # Release flock (use fixed FD 200 for bash 3.2 compatibility)
+            flock -u 200 2>/dev/null || true
+            exec 200>&- 2>/dev/null || true
+        elif [[ "$LOCK_ACQUIRED" == "file" ]]; then
+            # Remove file-based lock
+            rm -f "$LOCK_FILE" 2>/dev/null || true
+        fi
+    fi
+
+    exit $exit_code
+}
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)